File: benchmarks/fib/fib.c
#include <stdio.h>
#include <stdint.h>

int64_t fib(int64_t n) {
    int64_t a = 0, b = 1;
    for (int64_t i = 0; i < n; i++) {
        int64_t tmp = a + b;
        a = b;
        b = tmp;
    }
    return b;
}

int main() {
    printf("Result: %lld\n", (long long)fib(10000000));
    return 0;
}

File: benchmarks/fib/fib.java
class fib {
    static int fibInt(int n) {
        int a = 0, b = 1;
        for (int i = 0; i < n; i++) {
            int tmp = a + b;
            a = b;
            b = tmp;
        }
        return b;
    }
    public static void main(String[] args) {
        System.out.println("Result: " + fibInt(500000000));
    }
}
